xdsl/transforms/convert_ptr_to_riscv.py

```diff
@@ -0,0 +1,133 @@
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    TypeConversionPattern,
+    attr_type_rewrite_pattern,
+    op_type_rewrite_pattern,
+)
+
+
+class PtrTypeConversion(TypeConversionPattern):
+    @attr_type_rewrite_pattern
+    def convert_type(self, typ: ptr.PtrType) -> IndexType:
+        return IndexType()
+
+
+@dataclass
+class ConvertPtrAddOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.PtrAddOp, rewriter: PatternRewriter, /):
+        rewriter.replace_matched_op(arith.Addi(op.operands[0], op.operands[1]))
+
+
+@dataclass
+class ConvertStoreOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.StoreOp, rewriter: PatternRewriter, /):
+        addr, value = cast_operands_to_regs(rewriter)
+
+        match value.type:
+            case riscv.IntRegisterType():
+                new_op = riscv.SwOp(
+                    addr, value, 0, comment="store int value to pointer"
+                )
+            case riscv.FloatRegisterType():
+                float_type = cast(AnyFloat, op.value.type)
+                match float_type:
+                    case Float32Type():
+                        new_op = riscv.FSwOp(
+                            addr,
+                            value,
+                            0,
+                            comment="store float value to pointer",
+                        )
+                    case Float64Type():
+                        new_op = riscv.FSdOp(
+                            addr,
+                            value,
+                            0,
+                            comment="store double value to pointer",
+                        )
+                    case _:
+                        assert False, f"Unexpected floating point type {float_type}"
+
+            case _:
+                assert False, f"Unexpected register type {op.value.type}"
+
+        rewriter.replace_matched_op(new_op)
+
+
+@dataclass
+class ConvertLoadOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.LoadOp, rewriter: PatternRewriter, /):
+        casted = cast_operands_to_regs(rewriter)
+        addr = casted[0]
+
+        result_register_type = register_type_for_type(op.res.type)
+
+        match result_register_type:
+            case riscv.IntRegisterType:
+                lw_op = riscv.LwOp(addr, 0, comment="load word from pointer")
+            case riscv.FloatRegisterType:
+                float_type = cast(AnyFloat, op.res.type)
+                match float_type:
+                    case Float32Type():
+                        lw_op = riscv.FLwOp(addr, 0, comment="load float from pointer")
+                    case Float64Type():
+                        lw_op = riscv.FLdOp(addr, 0, comment="load double from pointer")
+                    case _:
+                        assert False, f"Unexpected floating point type {float_type}"
```

Comment:
similarly here, would be good to add a test here, since we know that there are other existing floating point types